autotest.pm

```diff
@@ -361,54 +362,60 @@ sub runalltests {
-                }
-                if ($bmwqemu::vars{DUMP_MEMORY_ON_FAIL}) {
-                    query_isotovideo('backend_save_memory_dump', {filename => $fullname});
-                }
-                if ($t->{fatal_failure} || $flags->{fatal} || (!exists $flags->{fatal} && !$snapshots_supported) || $bmwqemu::vars{TESTDEBUG}) {
-                    bmwqemu::stop_vm();
-                    return 0;
-                }
-                elsif (!$flags->{no_rollback} && $last_milestone) {
-                    load_snapshot('lastgood');
-                    $last_milestone->rollback_activated_consoles();
-                }
+        my $name = $t->{name};
+        bmwqemu::modstart "starting $name $t->{script}";
+        $t->start();
+
+        # avoid erasing the good vm snapshot
+        if ($snapshots_supported && (($bmwqemu::vars{SKIPTO} || '') ne $fullname) && $bmwqemu::vars{MAKETESTSNAPSHOTS}) {
+            make_snapshot($t->{fullname});
+        }
+
+        eval { $t->runtest; };
+        my $error = $@;    # save $@, it might be overwritten
+        $t->save_test_result();
+
+        if ($error) {
+            my $msg = $error;
+            if ($msg !~ /^test.*died/) {
+                # avoid duplicating the message
+                bmwqemu::diag $msg;
             }
-            else {
-                if (!$flags->{no_rollback} && $last_milestone && $flags->{always_rollback}) {
-                    load_snapshot('lastgood');
-                    $last_milestone->rollback_activated_consoles();
-                }
-                if ($snapshots_supported && ($flags->{milestone} || $bmwqemu::vars{TESTDEBUG})) {
-                    make_snapshot('lastgood');
-                    $last_milestone         = $t;
-                    $last_milestone_console = $selected_console;
-                }
+            if ($bmwqemu::vars{DUMP_MEMORY_ON_FAIL}) {
+                query_isotovideo('backend_save_memory_dump', {filename => $fullname});
+            }
+            if ($t->{fatal_failure} || $flags->{fatal} || (!exists $flags->{fatal} && !$snapshots_supported) || $bmwqemu::vars{TESTDEBUG}) {
+                bmwqemu::stop_vm();
+                return 0;
+            }
+            elsif (!$flags->{no_rollback} && $last_milestone) {
+                load_snapshot('lastgood');
+                $last_milestone->rollback_activated_consoles();
             }
         }
         else {
-            bmwqemu::diag "skipping $fullname";
-            $t->skip_if_not_running();
-            $t->save_test_result();
+            if (!$flags->{no_rollback} && $last_milestone && $flags->{always_rollback}) {
+                load_snapshot('lastgood');
+                $last_milestone->rollback_activated_consoles();
+            }
+            my $makesnapshot = $bmwqemu::vars{TESTDEBUG};
+            # Only make a snapshot if there is a next test and it's not a fatal milestone
+            if ($testindex ne $#testorder) {
```

Comment:
http://10.160.67.86/tests/723/file/autoinst-log.txt says

```
Can't call method "test_flags" on an undefined value at /usr/lib/os-autoinst/autotest.pm line 410.
```

Not sure if that's really the same line "410" here but I have not seen this error in before so I assume your changes caused this. That should be fixed.

Reply:
It's fixed after I changed the condition, but I didn't update the link to the verification runs.

Reply:
I don't have one for after the autotest addition yet, will add once they've finished.

Reply:
Done